examples/box_game/box_game.rs

```diff
@@ -50,9 +50,8 @@ pub struct Velocity {
     pub z: f32,
 }
 
-// You can also register resources. If your Component / Resource implements Hash, you can make use of `#[reflect(Hash)]`
-// in order to allow a GGRS `SyncTestSession` to construct a checksum for a world snapshot
-#[derive(Default, Reflect, Hash, Component)]
+// You can also register resources.
+#[derive(Resource, Default, Reflect, Hash)]
```

Comment:
I'm not sure if reflect/hash is still needed, I didn't check

Reply:
My understanding of this is still kind of shallow, but `register_rollback_resource` seems to need Reflect because it's needed for `TypeRegistration::insert`...

Or more specifically `TypeRegistration::insert` needs the `TypeData` trait to be implemented, which is automatically implemented by deriving `Reflect`.

I think `#[derive(Hash)]` and `#[reflect(Hash)` is not needed. At least the examples, including synctest, run fine without them. Perhaps @gschup would like to confirm?

Reply:
Deriving and reflecting hash is useful [here](https://github.com/Vrixyz/bevy_ggrs/blob/1cc28cce4d3b5e3216b893dc917fdacf7f37048b/src/world_snapshot.rs#L120). It is definitely optional.

Reply:
I guess we could be more specific about it in the comment... Something like "By deriving `Hash` and adding `reflect(Hash)` the resource can automatically be checked for desyncs during a `SyncTestSession`"?

Reply:
There is definitely a lot to be improved regarding documentation (see e.g. #35). It might probably even be better to that outside of this PR, in a big documentation commit.